skorch/_doctor.py

```diff
@@ -0,0 +1,936 @@
+        try:
+            self.net.partial_fit(X, y, **fit_params)
+        finally:
+            self._clean_up()
+
+        self.num_steps_ = len(self.activation_logs_[self.module_names_[0]])
+        self.fitted_ = True
+        return self
+
+    def get_layer_names(self):
+        """Return the names of all layers/modules
+
+        Returns
+        -------
+        names : dict of list of str
+          For each top level module, all layer names as a list of strings.
+
+        """
+        self.check_is_fitted()
+        names = {}
+        for module in self.module_names_:
+            if self.activation_logs_[module]:
+                names[module] = list(self.activation_logs_[module][0].keys())
+            else:
+                names[module] = []
+        return names
+
+    def get_param_names(self):
+        """Return all learnable parameters of the net
+
+        Returns
+        -------
+        names : dict of list of str
+          For each top level module, all parameter names as a list of strings.
+
+        """
+        self.check_is_fitted()
+        names = {}
+        for module in self.module_names_:
+            if self.gradient_logs_[module]:
+                # using the reversed order because gradients are recorded from
+                # last to first, but first to last is more intuitive to show
+                # TODO: When dropping python 3.7, dict keys are reversible, so
+                # no need to call list(keys)
+                keys = list(self.gradient_logs_[module][0].keys())
+                names[module] = list(reversed(keys))
+            else:
+                names[module] = []
+        return names
+
+    def predict(self, X, **kwargs):
+        """Calls the ``predict`` method of the underlying net"""
+        return self.net.predict(X, **kwargs)
+
+    def predict_proba(self, X, **kwargs):
+        """Calls the ``predict_proba`` method of the underlying net"""
+        return self.net.predict_proba(X, **kwargs)
+
+    def score(self, X, y=None, **kwargs):
+        """Calls the ``score`` method of the underlying net"""
+        return self.net.score(X, y=y, **kwargs)
+
+    def _get_axes(self, axes=None, figsize=None, nrows=1, squeeze=False):
+        """Helper function to get empty axes for plotting"""
+        if axes is not None:
+            return axes
+
+        try:
+            import matplotlib.pyplot as plt
+        except ImportError as exc:
+            msg = (
+                "This feature requires matplotlib to be installed; "
+                "please install it first, e.g. using "
+                "'python -m pip install matplotlib'"
+            )
+            raise ImportError(msg) from exc
+
+        width = 12
+        height = 6
+
+        if figsize is None:
+            figsize = (width, nrows * height)
+
+        _, axes = plt.subplots(nrows, 1, figsize=figsize, squeeze=squeeze)
+        return axes
+
+    def plot_loss(self, ax=None, figsize=None, **kwargs):
```

Comment:
Do you think it's useful to have two `kwargs` to pass into the train and val loss separately?